In [1]:
import numpy as np
import pandas as pd 
import pandas_profiling
import matplotlib.pyplot as plt 
import seaborn as sns
%matplotlib inline


**Loading Dataset**

In [2]:
airbnb= pd.read_csv("https://raw.githubusercontent.com/satyam007-aws/Airbnb-Price-Analysis/main/Listing%20copy%20(1).csv")
airbnb.shape

(6809, 21)

In [3]:
airbnb_copy=airbnb.copy()

**Understanding Data**

In [4]:
airbnb_copy.shape

(6809, 21)

In [5]:
airbnb_copy.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_cleansed',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bathroom',
       'Count of Amenities', 'price', 'availability_30', 'availability_60',
       'availability_90', 'availability_365', 'minimum_nights',
       'maximum_nights', 'number_of_reviews', 'review_scores_rating',
       'license'],
      dtype='object')

In [6]:
airbnb_copy.head(5)

id                                               name  host_id  \
0  528022.0                                     Somewhere Else  2594559   
1    2818.0          Quiet Garden View Room & Super Fast Wi-Fi     3159   
2   20168.0       Studio with private bathroom in the centre 1    59484   
3   27886.0  Romantic, stylish B&B houseboat in canal district    97647   
4   28871.0                            Comfortable double room   124245   

   host_name                  neighbourhood_cleansed  latitude  longitude  \
0        Els                              Westerpark  52.39114    4.88846   
1     Daniel  Oostelijk Havengebied - Indische Buurt  52.36435    4.94358   
2  Alexander                            Centrum-Oost  52.36407    4.89393   
3       Flip                            Centrum-West  52.38761    4.89188   
4      Edwin                            Centrum-West  52.36775    4.89092   

      room_type  accommodates bathroom  ...    price availability_30  \
0  Private room             2      1.5  ...  $100.00               4   
1  Private room             2      1.5  ...   $59.00               6   
2  Private room             2        1  ...  $106.00               0   
3  Private room             2      1.5  ...  $140.00               0   
4  Private room             2        1  ...   $75.00               0   

   availability_60  availability_90  availability_365  minimum_nights  \
0               16               29               212               2   
1               27               27                27               3   
2                0                0                 0               1   
3               17               32                40               2   
4                3                8                91               2   

   maximum_nights  number_of_reviews  review_scores_rating  \
0              30                248                  4.85   
1              28                314                  4.89   
2             365                339                  4.44   
3             356                243                  4.94   
4            1825                454                  4.88   

                    license  
0  0363 9289 A94D 5C21 A579  
1  0363 5F3A 5684 6750 D14D  
2  0363 CBB3 2C10 0C2A 1E29  
3  0363 974D 4986 7411 88D8  
4  0363 607B EA74 0BD8 2F6F  

[5 rows x 21 columns]

In [7]:
airbnb_copy.tail(5)

id                                               name  host_id  \
6804  794322.0                   Romantic Houseboat in the center  4184910   
6805  800723.0                  mediterranean ambiance appartment  4215174   
6806  801544.0                         Charlie + Coco's Apartment  1657181   
6807  802052.0  Sleeping in a unique ship in the center of A'dam!  4221960   
6808  805330.0         Sunny & Bright 3 bed apt with Roof terrace  4239235   

     host_name   neighbourhood_cleansed   latitude  longitude  \
6804     Lydia             Centrum-Oost  52.368637   4.899096   
6805     Hinde   De Baarsjes - Oud-West  52.368840   4.861270   
6806   Charlie  De Pijp - Rivierenbuurt  52.352050   4.902670   
6807     Peter             Centrum-Oost  52.364760   4.909250   
6808    Felice  De Pijp - Rivierenbuurt  52.347250   4.893410   

            room_type  accommodates bathroom  ...    price availability_30  \
6804     Private room             2        1  ...  $165.00              10   
6805  Entire home/apt             4        1  ...  $150.00               0   
6806  Entire home/apt             3        1  ...  $160.00               0   
6807     Private room             2        1  ...  $125.00               1   
6808  Entire home/apt             6      1.5  ...  $249.00               4   

      availability_60  availability_90  availability_365  minimum_nights  \
6804               33               55               264               1   
6805                0                0                 2               4   
6806                0                0                 0               3   
6807                9               21               133               1   
6808                4                4                 4               4   

      maximum_nights  number_of_reviews  review_scores_rating  \
6804            1125                343                  4.76   
6805              14                 18                  4.83   
6806            1124                 97                  4.57   
6807              30               1060                  4.93   
6808              21                 17                  4.94   

                       license  
6804  0363 7FCD 7E75 8341 EB0E  
6805  0363 6CD0 61EE A298 8E86  
6806  0363 B7E4 8394 0A13 58F5  
6807  0363 787E FE88 DA65 E7FC  
6808  0363 9611 7834 EC7F E5BB  

[5 rows x 21 columns]

In [8]:
airbnb_copy.describe()

id       host_id     latitude    longitude  accommodates  \
count  6.809000e+03  6.809000e+03  6809.000000  6809.000000   6809.000000   
mean   1.543794e+17  9.757724e+07    52.366847     4.890363      2.918784   
std    2.861164e+17  1.315182e+08     0.017026     0.035550      1.401175   
min    2.818000e+03  3.159000e+03    52.290340     4.755710      0.000000   
25%    1.510977e+07  9.287585e+06    52.356040     4.866890      2.000000   
50%    3.373018e+07  3.321458e+07    52.366190     4.888210      2.000000   
75%    5.313791e+07  1.336823e+08    52.376630     4.908470      4.000000   
max    7.740000e+17  4.901557e+08    52.425120     5.026430     16.000000   

       Count of Amenities  availability_30  availability_60  availability_90  \
count         6809.000000      6809.000000      6809.000000      6809.000000   
mean            30.210310         5.336320        13.703334        22.288295   
std             14.747885         7.751027        17.595904        28.135638   
min              0.000000         0.000000         0.000000         0.000000   
25%             19.000000         0.000000         0.000000         0.000000   
50%             29.000000         1.000000         4.000000         7.000000   
75%             40.000000         8.000000        25.000000        41.000000   
max             66.000000        30.000000        60.000000        90.000000   

       availability_365  minimum_nights  maximum_nights  number_of_reviews  \
count       6809.000000     6809.000000     6809.000000        6809.000000   
mean          84.314584        4.600676      443.405052          48.000147   
std          114.511750       30.446022      495.907373          99.425111   
min            0.000000        1.000000        1.000000           0.000000   
25%            0.000000        2.000000       21.000000           3.000000   
50%           22.000000        3.000000      100.000000          12.000000   
75%          140.000000        4.000000     1125.000000          43.000000   
max          365.000000     1001.000000     1825.000000        1875.000000   

       review_scores_rating  
count           6161.000000  
mean               4.817711  
std                0.273022  
min                0.000000  
25%                4.750000  
50%                4.890000  
75%                5.000000  
max                5.000000

In [9]:
airbnb_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6809 entries, 0 to 6808
Data columns (total 21 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      6809 non-null   float64
 1   name                    6809 non-null   object 
 2   host_id                 6809 non-null   int64  
 3   host_name               6809 non-null   object 
 4   neighbourhood_cleansed  6809 non-null   object 
 5   latitude                6809 non-null   float64
 6   longitude               6809 non-null   float64
 7   room_type               6809 non-null   object 
 8   accommodates            6809 non-null   int64  
 9   bathroom                6809 non-null   object 
 10  Count of Amenities      6809 non-null   int64  
 11  price                   6809 non-null   object 
 12  availability_30         6809 non-null   int64  
 13  availability_60         6809 non-null   int64  
 14  availability_90         6809 non-null   

In [10]:
airbnb_copy['price']=airbnb_copy['price'].str[1:]

In [11]:
airbnb_copy['price'].dtype

dtype('O')

In [12]:
airbnb_copy['price']=airbnb_copy['price'].str.replace(',','')

In [13]:
airbnb_copy['price']=airbnb_copy['price'].astype(float)

In [14]:
airbnb_copy['price'].dtype

dtype('float64')

In [15]:
airbnb_copy.describe(include="all")

id                    name       host_id host_name  \
count   6.809000e+03                    6809  6.809000e+03      6809   
unique           NaN                    6734           NaN      2799   
top              NaN  Apartment in Amsterdam           NaN     Peter   
freq             NaN                       8           NaN        41   
mean    1.543794e+17                     NaN  9.757724e+07       NaN   
std     2.861164e+17                     NaN  1.315182e+08       NaN   
min     2.818000e+03                     NaN  3.159000e+03       NaN   
25%     1.510977e+07                     NaN  9.287585e+06       NaN   
50%     3.373018e+07                     NaN  3.321458e+07       NaN   
75%     5.313791e+07                     NaN  1.336823e+08       NaN   
max     7.740000e+17                     NaN  4.901557e+08       NaN   

        neighbourhood_cleansed     latitude    longitude        room_type  \
count                     6809  6809.000000  6809.000000             6809   
unique                      22          NaN          NaN                4   
top     De Baarsjes - Oud-West          NaN          NaN  Entire home/apt   
freq                      1066          NaN          NaN             4910   
mean                       NaN    52.366847     4.890363              NaN   
std                        NaN     0.017026     0.035550              NaN   
min                        NaN    52.290340     4.755710              NaN   
25%                        NaN    52.356040     4.866890              NaN   
50%                        NaN    52.366190     4.888210              NaN   
75%                        NaN    52.376630     4.908470              NaN   
max                        NaN    52.425120     5.026430              NaN   

        accommodates bathroom  ...         price  availability_30  \
count    6809.000000     6809  ...   6809.000000      6809.000000   
unique           NaN       15  ...           NaN              NaN   
top              NaN        1  ...           NaN              NaN   
freq             NaN     4006  ...           NaN              NaN   
mean        2.918784      NaN  ...    231.185049         5.336320   
std         1.401175      NaN  ...    945.409907         7.751027   
min         0.000000      NaN  ...      0.000000         0.000000   
25%         2.000000      NaN  ...    125.000000         0.000000   
50%         2.000000      NaN  ...    180.000000         1.000000   
75%         4.000000      NaN  ...    256.000000         8.000000   
max        16.000000      NaN  ...  71536.000000        30.000000   

        availability_60  availability_90  availability_365  minimum_nights  \
count       6809.000000      6809.000000       6809.000000     6809.000000   
unique              NaN              NaN               NaN             NaN   
top                 NaN              NaN               NaN             NaN   
freq                NaN              NaN               NaN             NaN   
mean          13.703334        22.288295         84.314584        4.600676   
std           17.595904        28.135638        114.511750       30.446022   
min            0.000000         0.000000          0.000000        1.000000   
25%            0.000000         0.000000          0.000000        2.000000   
50%            4.000000         7.000000         22.000000        3.000000   
75%           25.000000        41.000000        140.000000        4.000000   
max           60.000000        90.000000        365.000000     1001.000000   

        maximum_nights  number_of_reviews  review_scores_rating  license  
count      6809.000000        6809.000000           6161.000000     6671  
unique             NaN                NaN                   NaN     5662  
top                NaN                NaN                   NaN   Exempt  
freq               NaN                NaN                   NaN      567  
mean        443.405052          48.000147              4.817711      NaN  
std         

In [16]:
airbnb_copy.price[airbnb_copy.price==0].count()

6

In [17]:
airbnb_copy.corr()

id   host_id  latitude  longitude  accommodates  \
id                    1.000000  0.224343 -0.033578  -0.028093     -0.004987   
host_id               0.224343  1.000000 -0.009644  -0.022449     -0.026615   
latitude             -0.033578 -0.009644  1.000000  -0.099020      0.064456   
longitude            -0.028093 -0.022449 -0.099020   1.000000      0.091596   
accommodates         -0.004987 -0.026615  0.064456   0.091596      1.000000   
Count of Amenities   -0.005120 -0.124008 -0.019149   0.036151      0.156538   
price                 0.020853 -0.005127  0.006151  -0.017821      0.082751   
availability_30       0.154912  0.241292 -0.003846  -0.027433     -0.026360   
availability_60       0.120770  0.242135 -0.000638  -0.011307     -0.034859   
availability_90       0.098607  0.233714 -0.002579  -0.004252     -0.037962   
availability_365      0.088108  0.226537  0.014602   0.000073     -0.012507   
minimum_nights       -0.008026 -0.009232 -0.001498   0.003461      0.003891   
maximum_nights       -0.211889 -0.042812  0.020255   0.018253     -0.005830   
number_of_reviews    -0.239476 -0.093631  0.025201  -0.002730     -0.080965   
review_scores_rating  0.042985 -0.159765  0.016389  -0.013474     -0.030744   

                      Count of Amenities     price  availability_30  \
id                             -0.005120  0.020853         0.154912   
host_id                        -0.124008 -0.005127         0.241292   
latitude                       -0.019149  0.006151        -0.003846   
longitude                       0.036151 -0.017821        -0.027433   
accommodates                    0.156538  0.082751        -0.026360   
Count of Amenities              1.000000  0.027599        -0.052134   
price                           0.027599  1.000000         0.014960   
availability_30                -0.052134  0.014960         1.000000   
availability_60                -0.045414  0.022708         0.870424   
availability_90                -0.045067  0.028411         0.794935   
availability_365               -0.063125  0.038418         0.585927   
minimum_nights                  0.003043 -0.000658         0.048898   
maximum_nights                 -0.038294 -0.013680         0.034244   
number_of_reviews               0.026711 -0.015910         0.053332   
review_scores_rating            0.162775 -0.001407        -0.145660   

                      availability_60  availability_90  availability_365  \
id                           0.120770         0.098607          0.088108   
host_id                      0.242135         0.233714          0.226537   
latitude                    -0.000638        -0.002579          0.014602   
longitude                   -0.011307        -0.004252          0.000073   
accommodates                -0.034859        -0.037962         -0.012507   
Count of Amenities          -0.045414        -0.045067         -0.063125   
price                        0.022708         0.028411          0.038418   
availability_30              0.870424         0.794935          0.585927   
availability_60              1.000000         0.964870          0.715270   
availability_90              0.964870         1.000000          0.765003   
availability_365             0.715270         0.765003          1.000000   
minimum_nights               0.029660         0.023312          0.002319   
maximum_nights               0.050525         0.057802          0.059577   
number_of_reviews            0.130971         0.156663          0.178641   
review_scores_rating        -0.164915        -0.164327         -0.170254   

                      minimum_nights  maximum_nights  number_of_reviews  \
id                         -0.008026       -0.211889          -0.239476   
host_id                    -0.009232       -0.042812          -0.093631   
latitude                   -0.001498        0.020255           0.025201   
longitude                   0.003461        0.018253          -0.002730   
accommodates                0.0038

In [18]:
airbnb_copy['neighbourhood_cleansed'].value_counts().head(5)

De Baarsjes - Oud-West     1066
Centrum-West                930
De Pijp - Rivierenbuurt     760
Centrum-Oost                680
Westerpark                  468
Name: neighbourhood_cleansed, dtype: int64

In [19]:
airbnb_copy['room_type'].value_counts()

Entire home/apt    4910
Private room       1798
Hotel room           62
Shared room          39
Name: room_type, dtype: int64

In [20]:
Total=(airbnb_copy['room_type'].value_counts()/len(airbnb_copy))*100
Total

Entire home/apt    72.110442
Private room       26.406227
Hotel room          0.910560
Shared room         0.572771
Name: room_type, dtype: float64

**72 % of data Belong to Entire home/apt , 26% of data belong to Private room**

**Airbnb Majorly Deal with Entire home/apt room type in Amsterdam**

### Preprocessing

In [21]:
miss =  airbnb_copy.isnull().sum()
miss1 = (airbnb_copy.isnull().sum()/len(airbnb_copy))*100
miss_data = pd.concat([miss,miss1],axis=1,keys=["Total","%"])
miss_data

Total         %
id                          0  0.000000
name                        0  0.000000
host_id                     0  0.000000
host_name                   0  0.000000
neighbourhood_cleansed      0  0.000000
latitude                    0  0.000000
longitude                   0  0.000000
room_type                   0  0.000000
accommodates                0  0.000000
bathroom                    0  0.000000
Count of Amenities          0  0.000000
price                       0  0.000000
availability_30             0  0.000000
availability_60             0  0.000000
availability_90             0  0.000000
availability_365            0  0.000000
minimum_nights              0  0.000000
maximum_nights              0  0.000000
number_of_reviews           0  0.000000
review_scores_rating      648  9.516816
license                   138  2.026729

In [22]:
airbnb_copy.price[airbnb_copy.price==0].count()

6

**Droping entries with Price <=0**

In [23]:
airbnb_copy=airbnb_copy.drop(airbnb_copy.price[airbnb_copy.price <=0].index)

In [24]:
airbnb_copy.price[airbnb_copy.price==0].count()

0

In [25]:
airbnb_copy.duplicated().sum()

0

Their is no duplicated rows

In [26]:
airbnb_copy['review_scores_rating'].value_counts()

5.00    1975
4.88     179
4.89     154
4.86     152
4.75     151
        ... 
4.15       1
3.43       1
4.11       1
4.02       1
4.06       1
Name: review_scores_rating, Length: 118, dtype: int64

In [27]:
airbnb_copy.groupby(['review_scores_rating'])['price'].count()

review_scores_rating
0.00       4
2.00       4
2.67       1
2.75       1
3.00       4
        ... 
4.96      95
4.97      88
4.98      64
4.99       9
5.00    1975
Name: price, Length: 118, dtype: int64

In [28]:
airbnb_copy['review_scores_rating'].isnull().sum()

644

**Their are 644 null values in review_scores_rating so here i will replace the NAN with zero reviews**

In [29]:
airbnb_copy['review_scores_rating']=airbnb_copy['review_scores_rating'].fillna(0)

In [30]:
airbnb_copy['review_scores_rating'].isnull().sum()

0

In [31]:
airbnb_copy['bathroom'].value_counts()

1            4006
1.5          2094
2             386
2.5           146
3              84
0              34
3.5            23
4               6
Half-bath       5
5               5
4.5             5
Private         4
Shared          2
5.5             2
17              1
Name: bathroom, dtype: int64

In [32]:
airbnb_copy['bathroom'].dtype

dtype('O')

In [33]:
airbnb_copy['bathroom'][airbnb_copy.bathroom.isin(["Half-bath","Private","Shared"])].count()

11

In [34]:
airbnb_copy['bathroom']=airbnb_copy['bathroom'].replace(["Half-bath","Private","Shared"],1)

In [35]:
airbnb_copy['bathroom']=airbnb_copy['bathroom'].astype(float)

In [36]:
airbnb_copy['bathroom'].dtype

dtype('float64')

In [37]:
airbnb_copy['bathroom'].median()

1.0

In [38]:
airbnb_copy["booked_days"]=365-airbnb_copy["availability_365"]

In [39]:
airbnb_copy["earning_till_date"]=airbnb_copy["booked_days"]*airbnb_copy["price"]

In [40]:
airbnb_copy['earning_till_date'].dtype

dtype('float64')

In [41]:
airbnb_copy.head(3)

id                                          name  host_id  host_name  \
0  528022.0                                Somewhere Else  2594559        Els   
1    2818.0     Quiet Garden View Room & Super Fast Wi-Fi     3159     Daniel   
2   20168.0  Studio with private bathroom in the centre 1    59484  Alexander   

                   neighbourhood_cleansed  latitude  longitude     room_type  \
0                              Westerpark  52.39114    4.88846  Private room   
1  Oostelijk Havengebied - Indische Buurt  52.36435    4.94358  Private room   
2                            Centrum-Oost  52.36407    4.89393  Private room   

   accommodates  bathroom  ...  availability_60  availability_90  \
0             2       1.5  ...               16               29   
1             2       1.5  ...               27               27   
2             2       1.0  ...                0                0   

   availability_365  minimum_nights  maximum_nights  number_of_reviews  \
0               212               2              30                248   
1                27               3              28                314   
2                 0               1             365                339   

   review_scores_rating                   license  booked_days  \
0                  4.85  0363 9289 A94D 5C21 A579          153   
1                  4.89  0363 5F3A 5684 6750 D14D          338   
2                  4.44  0363 CBB3 2C10 0C2A 1E29          365   

   earning_till_date  
0            15300.0  
1            19942.0  
2            38690.0  

[3 rows x 23 columns]

In [45]:
airbnb_copy.shape

(6803, 23)